libursa/tests/cl.rs

```diff
@@ -1174,7 +1174,7 @@ mod cl_tests {
                 )
                 .unwrap();
 
-            assert!(proof_verifier.verify(&proof, &nonce).unwrap());
+            assert!(!proof_verifier.verify(&proof, &nonce).unwrap());
```

Comment:
Why is this test condition flipped?

Reply:
The proof was created with 2 credentials with different link secrets and hence was expected to fail.

Reply:
Thanks for finding that defect.
It looks like it's corrected over three unrelated commits in this PR. Could you please organize the fix into one commit?
Also I'm not clear why this test wasn't preventing CI from passing earlier? I don't see an ignore or anything like that.

Reply:
My first 3 commits were for producing a broken test that illustrated the problem (and some comments). My next commit fixed the issue. Then there are some FFI additions and other changes to address the code review. I would suggest to squash all commits into 2, the first illustrates the issue with a failing test and the second with the fix, FFI included.
As to why the CI did not fail earlier, i had not raised the PR until i did the fix and i don't think CI runs on each commit.

Reply:
@dcmiddle Are you fine with the way commits are laid out now?

Reply:
Can you please update the commit messages? I think what this PR is about is there was a defect in the anon credentials code, the unit test which should have exposed the problem had the wrong assertion set so that the test would pass, and you fixed the core issue, corrected the test and added a new test. Whether I got that right or not can you please update the commit messages so the actual story is clearer? All this is obvious when you are the person who spent hours/days/weeks on something and it's entirely not obvious to other people, especially contributors a year from now that will try to figure out the code. :)
This post is helpful:
https://chris.beams.io/posts/git-commit/#seven-rules

Reply:
Done